mesonbuild/interpreter/type_checking.py

```diff
@@ -548,6 +548,12 @@ def link_whole_validator(values: T.List[T.Union[StaticLibrary, CustomTarget, Cus
         validator=in_set_validator({'', 'default', 'internal', 'hidden' 'protected', 'inlineshidden'}),
     ),
     KwargInfo('install_rpath', str, default=''),
+    KwargInfo(
+        'link_depends',
+        ContainerTypeInfo(list, (str, File, CustomTarget, CustomTargetIndex)),
```

Comment:
It is, unfortunately, legal to pass e.g. a build.Executable here, since it has a `.get_outputs()` method.

This is, naturally, contrary to the code comments and the error message emitted if you pass something without such a function to it. Nonetheless, it works today as far as meson is concerned.

I won't get into the topic of what linker arguments are theoretically capable of doing something with an executable program at link time.